### PR TITLE
fix: correctly expand derives for namespaced JSON outputs

### PR DESCRIPTION
**Pull request by klkvr**
## Motivation

Addresses https://github.com/alloy-rs/core/pull/694#issuecomment-2249896421

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/sol-macro-input/src/json.rs [hunk 1]
```diff
@@ -19,10 +19,39 @@ impl SolInput {
...
+
+        let library_tokens = library_tokens_iter.by_ref();
+
+        let mut libraries = Vec::new();
+
+        while library_tokens.peek().is_some() {
+            let sol_library_tokens: TokenStream = std::iter::once(TokenTree::Ident(id("library")))
+                .chain(
+                    library_tokens
+                        .take_while(|tt| !matches!(tt, TokenTree::Ident(id) if id == "library")),
```
Comment (mattsse) [inline crates/sol-macro-input/src/json.rs:44]: could this end up looping if `library_tokens.take_while` does not consume a value?

Comment (klkvr) [reply crates/sol-macro-input/src/json.rs:44]: It should just consume all left elements if closure never returns `true`

Comment (mattsse) [reply crates/sol-macro-input/src/json.rs:44]: right

**Review by mattsse — APPROVED** [review]
lgtm,

clippy is unrelated

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**